### ISSUE TITLE
Make Unit Tests Work

### ISSUE THREAD

**Issue by ilan-gold** [theislab/batchglm#119]
Currently the unit tests fail for a variety of reasons.  These can be confined to three main issues:

1. Tensorflow issues, like TF1-dask compatibility, convergence or unimplemented TF2 - we should remove both while waiting on TF2 to be merged
2. Code structure issues like missing modules or incorrect paths
3. Bugs in actual functions

I think two separate PR's, one for the first problem and then 1 for the second two is a clean way to deal with this.

I will remove TF1 (to be replaced with TF2) in the first PR and then handle the rest in the second.
